src/main/java/com/dani/blog/config/BlogWebAppInitializer.java

```diff
@@ -0,0 +1,37 @@
+package com.dani.blog.config;
+
+
+import org.springframework.web.servlet.support.AbstractAnnotationConfigDispatcherServletInitializer;
+
+/**
+ * Configuration alternative to web.xml.
+ *
+ * Works with ServletContainerInitializer (introduced in Servlet 3.0 at 2009).
+ * @author dani
+ */
+public class BlogWebAppInitializer extends AbstractAnnotationConfigDispatcherServletInitializer {
```

Comment:
It's from the Servlet 3.0 specs, it's scanning all the classes in the classpath if they implement `ServletContainerInitializer` (`AbstractAnnotationConfigDispatcherServletInitializer` does) and then automatically asks that to define the context and the rest. Less xml!